### DISCUSSION TITLE
Assioma connection

### DISCUSSION THREAD

**Discussion by posssson** [doudar/SmartSpin2k#290] [Q&A] [answered]
Hi everyone,

I am maybe the only one in this situation, but with after the installation of the last develop, I can't succed to connect my assioma pedals. I have only any and none in settings. I can find pedals on my phone but even if I cut bluetooth on the phone, it does'nt work. Any ideas?

**Comment by doudar** [comment]
I think you found a bug. https://github.com/doudar/SmartSpin2k/blob/9a1efa80566b737644be0d02484678f443ac75e0/data/bluetoothscanner.html#L207

needs to be updated to  xhttp.open("GET", "/runtimeConfigJSON", true);

and probably some other stuff on that .html as well. 

I'll get on this today. 

Without this setting though, unless you had them set to "none" previously, it should still connect if you ignore dropdowns and connect, meaning you should still see power data coming through the debug log.

**Reply by doudar** [reply]
Looking at the code, "any" and "none" should still be getting passed correctly. 

For now set "any" and it will work (at least behind the scenes). 

I think what I'm going to do is actually move the "foundDevices" back to "userParameters" as the fix because this might be good information to keep saved in the config.txt.

**Reply by posssson** [reply]
Indeed "any" find the power of assioma but zwift doesn't find a controllable home trainer. Don't know if it is reliable

**Reply by doudar** [reply]
Anytime you do a FW update, in Zwift you'll have to "unpair" the SS2K (from every field), then re-pair it as a powermeter, then after a few seconds it will also show up (selectable) as a controllable trainer.

**Reply by posssson** [reply]
Yes I did it :( I will try again

**Reply by doudar** [reply]
Sorry about that - no idea how Zwift can tell that it's a new FW but it does somehow. 

Keep me updated

**Reply by posssson** [reply]
So I can connect SS2 on zwift (windows) but not the controllable trainer. But everything works on my mac... Don't know why :).

**Reply by doudar** [reply]
The newest Develop should have the webpage fixed if you want to try it and report back. It also has auto-min-max which is still somewhat experimental and might require further tweaking. 

auto-min-max uses the new powertable to calculate the minimum and maximum travel for the stepper motor. Defaults are >50w and <700W. The high limit can be changed on the settings webpage.  

It also features settings to flip the direction of the stepper motor and shifter switches to compensate for wiring.

**Reply by posssson** [reply]
Yep it works! I can connect sensors. And my windows didn't find the controllable trainer between I paired the bluetooth of the windows with SS2 in settings and when I removed this connection everything works perfectly. Thank you.

**Comment by posssson** [comment]
Today new version of zwift and now motor doesn't turn.

**Reply by MarkusSchneider** [reply]
Have the same problem. Seems that erg_mode is always false.

**Reply by posssson** [reply]
:( 
Do you an idea of the workload to create an app to control the trainer without zwift? I think just a c++ app with a timer a power target and a configurable workout.

**Reply by MarkusSchneider** [reply]
the btsimulator page has on option to set Target Power manually for testing.
http://smartspin2k.local/btsimulator.html
![image](https://user-images.githubusercontent.com/7011066/147785294-866ac4c9-f2d1-4a16-a12a-6dc6f603f317.png)
You must enable ERG and ERG Target Watts, then you can set the required power via slider 😉

**Reply by posssson** [reply]
Ohhh nice I will code something to use this and don't use zwift.

**Reply by doudar** [reply]
Is this a temporary Zwift issue, or something we need to fix on our end?

**Reply by doudar** [reply]
@MarkusSchneider It looks like Zwift implemented something we either don't have or we aren't providing the response it would like. This repeats a bunch in the Zwift log.

[15:20:39] [FTMS] Updated state from 5 to 1 and request opcode from 1 to 127

[15:20:39] [FTMS] Request took too long, resetting trainer.

[15:20:39] [FTMS] Updated state from 1 to 4 and request opcode from 127 to 0

[15:20:40] NETCLIENT:[WARN] UDP connection timeout (4 so far), reconnection attempt 4

[15:20:40] NETCLIENT:[INFO] Connecting to UDP server...

[15:20:40] NETWORK:error (6) sending player state

[15:20:41] [FTMS] Updated state from 4 to 5 and request opcode from 0 to 1

**Reply by doudar** [reply]
fixed by Commit 7f3b217 

Thanks for reporting this guys! 

I'll try to get a new binary out tonight.

**Comment by doudar** [comment, marked as answer]
New Binary is live on the OTA update. 1.12.30

It fixes the Bluetooth page, adds auto min_max detection, fixes the recent Zwift issue, and fixed another small bug I found in the powertable logic.

**Reply by posssson** [reply]
Nice work !!

**Reply by MarkusSchneider** [reply]
Thx for fixing all the issues. Great work 👍
In FTMS specification there is also a control point for "Stop" specified (Op Code 0x08) don't know if we need this also. It's a mandatory service.
![image](https://user-images.githubusercontent.com/7011066/147814364-29727d61-1147-4b21-97c7-b0dfb3d14a40.png)


Greetings
Markus

**Reply by doudar** [reply]
It's easy to pop it in there for the next update.
